www/js/diary/list.js

```diff
@@ -487,16 +494,20 @@ angular.module('emission.main.diary.list',['ui-leaflet',
    {text:'Religious', value:'religious'},
    {text:'Other',value:'other_purpose'}];
 
-   $scope.storeMode = function(mode_val) {
+   $scope.storeMode = function(mode_val, isOther) {
       $scope.draftMode.label = mode_val;
       Logger.log("in storeMode, after setting mode_val = "+mode_val+", draftMode = "+JSON.stringify($scope.draftMode));
       $window.cordova.plugins.BEMUserCache.putMessage(MODE_CONFIRM_KEY, $scope.draftMode);
+      if(isOther = true) 
+        $scope.draftPurpose = angular.undefined;
    }
 
-   $scope.storePurpose = function(purpose_val) {
+   $scope.storePurpose = function(purpose_val, isOther) {
       $scope.draftPurpose.label = purpose_val;
       Logger.log("in storePurpose, after setting purpose_val = "+purpose_val+", draftPurpose = "+JSON.stringify($scope.draftPurpose));
       $window.cordova.plugins.BEMUserCache.putMessage(PURPOSE_CONFIRM_KEY, $scope.draftPurpose);
```

Comment:
doesn't this store the `draftPurpose` (which is `undefined` in this case) into the database?

Reply:
This gives an error because $scope.draftPurpose.label is undefined so it is not possible to assign purpose_val to it, this is the bug.

Reply:
@sunil07t But how does your change fix this? you pass in `isOther`, but you don't actually check it until the last line.

Reply:
This check is just to make `draftPurpose` undefined

Reply:
This is just to make the `draftPurpose` undefined

Reply:
I am sorry, I still don't understand how this works.

```
$scope.storePurpose = function(purpose_val, isOther) {
<----- here, $scope.draftPurpose is undefined
       $scope.draftPurpose.label = purpose_val; <--- so why doesn't this try to set a subfield of undefined?
       Logger.log("in storePurpose, after setting purpose_val = "+purpose_val+", draftPurpose = "+JSON.stringify($scope.draftPurpose));
```

Did you re-test after making the boolean check? Can you explain to me how it works?

Reply:
Yes, I re-tested it after fixing the boolean check. The boolean check just makes the `draftPurpose` undefined at the end because the closePopup functions will not make `draftPurpose` undefined and that is what we want. The closePopup functions makes `draftPurpose` undefined before the user has a chances to add the label.

Reply:
However, we need closePopover function to close the tab when the user presses one of the options in the popover list.